### PR TITLE
Locations, spaces, topics of ideas.

### LINKED ISSUE #165: Reduce redundancy in data model
In `Types`:

```haskell
data Idea = Idea
    [...]
    , _ideaSpace      :: IdeaSpace
    , _ideaTopic      :: Maybe (AUID Topic)
```

To avoid inconsistencies: 

- drop `_ideaSpace`, `_ideaTopic`
- replace with `_ideaLocation :: Either IdeaSpace (AUID Topic)`

### PR DESCRIPTION
**Pull request by fisx**
This raises some interesting questions on how we currently do stuff.  I think it was a good idea to clean this up!

Fixes #165

### CODE REVIEW

**Comment by fisx** [comment]
this should be relatively strait-forward.  one tricky thing is that so far ideas are only edited and created in ideaspaces, not in topics.  so some of the form paths / actions / whatnot need to be made smarter.

File: src/Types.hs [hunk 1]
```diff
@@ -93,6 +92,11 @@ data Idea = Idea
 
 instance SOP.Generic Idea
 
+newtype IdeaLocation = IdeaLocation { _fromIdeaLocation :: Either IdeaSpace (AUID Topic) }
```
Comment (andorp) [inline src/Types.hs]: Please use an ADT here.

**Comment by fisx** [comment]
I just realized that I have a newer version of this branch locally.  Can I rebase and force-push?

**Comment by fisx** [comment]
I opened #209 and will close this as obsoleted.

**Pull request closed without merging**